tsinfer/formats.py

```diff
@@ -2348,46 +2390,52 @@ def sites_metadata(self):
         try:
             return [
                 schema.decode_row(r)
-                for r in self.data["sites_metadata"][:][self.sites_mask]
+                for r in self.data["sites_metadata"][:][self.sites_select]
             ]
         except KeyError:
             return [{} for _ in range(self.num_sites)]
 
     @functools.cached_property
     def sites_time(self):
         try:
-            return self.data["sites_time"][:][self.sites_mask]
+            return self.data["sites_time"][:][self.sites_select]
         except KeyError:
             return np.full(self.num_sites, tskit.UNKNOWN_TIME)
 
     @functools.cached_property
     def sites_position(self):
-        return self.data["variant_position"][:][self.sites_mask]
+        return self.data["variant_position"][:][self.sites_select]
 
     @functools.cached_property
     def sites_alleles(self):
-        return self.data["variant_allele"][:][self.sites_mask]
+        return self.data["variant_allele"][:][self.sites_select]
 
     @functools.cached_property
-    def sites_mask(self):
-        try:
-            if (
-                self.data["variant_mask"].shape[0]
-                != self.data["variant_position"].shape[0]
-            ):
+    def sites_select(self):
+        if self._sites_mask_name is None:
+            return np.full(self.data["variant_position"].shape, True, dtype=bool)
+        else:
+            try:
```

Comment:
Basically all of the logic for this method could be moved to the ``__init__``, couldn't it? That would make it possible to catch these kinds of errors at init time rather than later on.

Why not also just store the ``sites_select`` array then rather than faffing with a cached_property? This is a read-only view, isn't it?

Reply:
Fixed in https://github.com/tskit-dev/tsinfer/pull/896/commits/56df80090c8fcb36bc18ca192f97df063d4f51b2